### PR TITLE
AMM Flag Updates

### LINKED ISSUE #2409: AMM Payment Flag
It's not obvious how to identify if a `Payment` transaction swapped with an AMM pool.

To check this, you have to look at the tx metadata and search for the AMM trustline flag to determine that a transaction interacted with it.

Following is partial metadata for a payment via AMM:

```
{'ModifiedNode': {'FinalFields': {'Balance': {'currency': 'USD',
                                                                                     'issuer': 'rrrrrrrrrrrrrrrrrrrrBZbvji',
                                                                                     'value': '-10010.01001001001'},
                                                                         'Flags': 16908288,
                                                                         'HighLimit': {'currency': 'USD',
                                                                                       'issuer': 'ammXRP-USD',
                                                                                       'value': '0'},
                                                                         'HighNode': '0',
                                                                         'LowLimit': {'currency': 'USD',
                                                                                      'issuer': 'gw',
                                                                                      'value': '0'},
                                                                         'LowNode': '0'},
                                                         'LedgerEntryType': 'RippleState',
                                                         'LedgerIndex': '7A699DC38D20550C7F2C3359D8AE85886AEDD8217A77A04DCB682D3EF927D60B',
                                                         'PreviousFields': {'Balance': {'currency': 'USD',
                                                                                        'issuer': 'rrrrrrrrrrrrrrrrrrrrBZbvji',
                                                                                        'value': '-10000'}},
                                                         'PreviousTxnID': '8262AD81183C3938E362379123BA28E56AB58810229987498BE24D4323445C19',
                                                         'PreviousTxnLgrSeq': 40}}]
```

Note Flags = `16908288`. AMM trustline has a flag `131072`. `16908288` & `131072` = `131072` ; i.e. this is AMM's trustline.

### LINKED ISSUE #2488: Add `lsfAMMNode` to RippleState Flags.
https://github.com/XRPLF/xrpl-dev-portal/blob/d42678b96f5b62cf56a554612f1cc054df70694f/docs/references/protocol/ledger-data/ledger-entry-types/ripplestate.md#L74-L83

https://github.com/XRPLF/rippled/blob/0c32fc5f2adb5a1598dd7e97fc83ba0f64fdee20/src/ripple/protocol/LedgerFormats.h#L299-L300

```cpp
    lsfAMMNode = 0x01000000,     // True, trust line to AMM. Used by client
                                 // apps to identify payments via AMM.
```

Thanks to @mvadari

### PR DESCRIPTION
**Pull request by oeggert**
Also added note how to find if a payment interacted with an AMM.

### CODE REVIEW

File: docs/concepts/tokens/decentralized-exchange/automated-market-makers.md [hunk 1]
```diff
@@ -29,6 +29,12 @@ An AMM holds two different assets: at most one of these can be XRP, and one or b
 
 When users want to trade in the decentralized exchange, their [Offers](offers.md) and [Cross-Currency Payments](../../payment-types/cross-currency-payments.md) can automatically use AMMs to complete the trade. A single transaction might execute by matching Offers, AMMs, or a mix of both, depending on what's cheaper.
 
+{% admonition type="info" name="Note" %}
+
+You can determine if a `Payment` interacted with an AMM by checking for a [`RippleState`](../../../references/protocol/ledger-data/ledger-entry-types/ripplestate.md) ledger entry in the transaction metadata. A flag value of `16777216` indicates AMM liquidity was consumed.
```
Comment (tequdev) [inline docs/concepts/tokens/decentralized-exchange/automated-market-makers.md]: Isn't it the same for the OfferCreate transaction as well as the Payment transaction?

Comment (oeggert) [reply docs/concepts/tokens/decentralized-exchange/automated-market-makers.md]: Good catch, adding it in.

**Review by DennisDawson — APPROVED** [review]
LGTM.

**Review by amarantha-k — APPROVED** [review]
lgtm

**Pull request merged**